src/proof/range_proof.rs

```diff
@@ -0,0 +1,374 @@
+use std::fmt;
+use std::iter;
+use std::str;
+use std::mem;
+use std::error::Error;
+use std::borrow::Borrow;
+use ring::digest::{Context, SHA256};
+use rand::random;
+use ::arithimpl::traits::*;
+use ::BigInteger as BigInt;
+use ::Paillier as Paillier;
+use ::core::*;
+use proof::correct_key::*;
+use ::traits::*;
+use ::{EncryptionKey, DecryptionKey};
+
+use proof::correct_key::ProofError;
+const STATISTICAL_ERROR_FACTOR: usize = 40;
+const STATISTICAL_ERROR_FACTOR_IN_BYTES: usize = 5;
+const SECURITY_PARAM : usize  = 256;
+const BITS_IN_BYTE: usize = 8;
```

Comment:
Magic number here, can you add a comment why 8?

Reply:
I cannot explain it better than bits in a byte